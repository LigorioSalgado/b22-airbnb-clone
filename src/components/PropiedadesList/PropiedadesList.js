import React, { Component } from 'react';
import PropiedadCard from './PropiedadCard/PropiedadCard';
import  gql from 'graphql-tag';
import {Query} from 'react-apollo';


const GET_HOUSES = gql`
    query{
        allPropiedades{
            _id,
            nombre,
            descripcion_corta,
            fotos,
            precio,
            calificacion{
                estrellas
            }

        }
    }
`;

class PropiedadesList extends Component{

    constructor(props){
        super(props);
    }

    redirectDetail = (id) => {
       this.props.history.push(`/propiedad/${id}`);
    }

    renderHouses = () => (
            <Query query={GET_HOUSES} >
                    {({loading,error,data}) => {
                        if(loading) return "Loading....";
                        if(error) return "No hay casas :";
                        if(data) console.log(data);

                        return(
                            <div className="row justify-content-center mt-5">
                               {data.allPropiedades.map((propiedad) => (
                                     <div className="col-md-6 col-lg-3">
                                            <PropiedadCard propiedad={propiedad}
                                                redirect={this.redirectDetail}
                                            />
                                    </div> 
                               ))} 

                            </div>
                        )

                    }}


            </Query>
    )

    render(){
        return this.renderHouses();
    }
}

export default PropiedadesList;