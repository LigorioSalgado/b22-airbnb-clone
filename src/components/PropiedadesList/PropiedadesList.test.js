import React from 'react';
import {shallow} from 'enzyme';

import PropiedadesList from './PropiedadesList';

describe("PropiedadesList",()=>{

    it("Debe renderear correctamente", () =>{

        const component =  shallow(<PropiedadesList/>)

        expect(component).toMatchSnapshot();
    })

})