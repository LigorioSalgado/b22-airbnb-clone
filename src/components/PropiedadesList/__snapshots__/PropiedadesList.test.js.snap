// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropiedadesList Debe renderear correctamente 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": undefined,
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "allPropiedades",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "_id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "nombre",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "descripcion_corta",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "fotos",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "precio",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "calificacion",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "estrellas",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 226,
        "start": 0,
      },
    }
  }
>
  <Component />
</Query>
`;
